### PR TITLE
[toMap-vs-groupingBy] toMap vs groupingBy

### PR DESCRIPTION
**Pull request by sk1418**

### CODE REVIEW

File: core-java-modules/core-java-streams-maps/src/test/java/com/baeldung/streams/tomapvsgroupingby/ToMapVsGroupingByUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    private City franceNull = new City(null, "France");
+    private City germanyNull = new City(null, "Germany");
+
+    @Test
+    void whenUsingToMap_thenGetExpectedResult() {
+        Map<String, City> result = Stream.of(paris, berlin, london, newYork, tokyo)
+            .collect(Collectors.toMap(City::getCountry, Function.identity()));
+
+        Map<String, City> expected = Map.of(
+            // @formatter:off
```
Comment (theangrydev) [inline core-java-modules/core-java-streams-maps/src/test/java/com/baeldung/streams/tomapvsgroupingby/ToMapVsGroupingByUnitTest.java:80]: What happens without `// @formatter:off`, does our auto formatter try to indent too much or something? It's a bit distracting having this syntax throughout the tests

File: core-java-modules/core-java-streams-maps/src/test/java/com/baeldung/streams/tomapvsgroupingby/ToMapVsGroupingByUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    private City london = new City("London", "UK");
+    private City newYork = new City("New York", "USA");
+    private City tokyo = new City("Tokyo", "Japan");
+
+    private City nice = new City("Nice", "France");
+    private City hamburg = new City("Hamburg", "Germany");
+    private City aachen = new City("Aachen", "Germany");
+
+    private City franceNull = new City(null, "France");
+    private City germanyNull = new City(null, "Germany");
```
Comment (theangrydev) [inline core-java-modules/core-java-streams-maps/src/test/java/com/baeldung/streams/tomapvsgroupingby/ToMapVsGroupingByUnitTest.java:72]: Can we make these private static final and uppercase if we are using them as constants? Or at least private final

**Review by theangrydev — CHANGES_REQUESTED** [review]

File: core-java-modules/core-java-streams-maps/src/test/java/com/baeldung/streams/tomapvsgroupingby/ToMapVsGroupingByUnitTest.java (hunk 1 above)
Comment (sk1418) [reply core-java-modules/core-java-streams-maps/src/test/java/com/baeldung/streams/tomapvsgroupingby/ToMapVsGroupingByUnitTest.java:80]: without "@formatter:off", all these parameters will be merged into one single line:

`Map<String, City> expected = Map.of("France", paris, "Germany", berlin, "UK", london, "USA", newYork, "Japan", tokyo);`

I think it's easier to read if we break the long line. 

BTW, those formatter-related comments are not in the article.

File: core-java-modules/core-java-streams-maps/src/test/java/com/baeldung/streams/tomapvsgroupingby/ToMapVsGroupingByUnitTest.java (hunk 2 above)
Comment (sk1418) [reply core-java-modules/core-java-streams-maps/src/test/java/com/baeldung/streams/tomapvsgroupingby/ToMapVsGroupingByUnitTest.java:72]: OK, I will make them final. They can be static for sure, but IMO, lowercase words are easier to read than UPPERCASE words. 
I will make the change f together with your article suggestions.

File: core-java-modules/core-java-streams-maps/src/test/java/com/baeldung/streams/tomapvsgroupingby/ToMapVsGroupingByUnitTest.java (hunk 1 above)
Comment (theangrydev) [reply core-java-modules/core-java-streams-maps/src/test/java/com/baeldung/streams/tomapvsgroupingby/ToMapVsGroupingByUnitTest.java:80]: Makes sense ok

File: core-java-modules/core-java-streams-maps/src/test/java/com/baeldung/streams/tomapvsgroupingby/ToMapVsGroupingByUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+    public String getCountry() {
+        return country;
+    }
+
+    @Override
+    public final boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof City city)) {
```
Comment (theangrydev) [inline core-java-modules/core-java-streams-maps/src/test/java/com/baeldung/streams/tomapvsgroupingby/ToMapVsGroupingByUnitTest.java:44]: Maybe rewrite this without the Java 16 pattern matching so it works with lower Java versions too?

File: core-java-modules/core-java-streams-maps/src/test/java/com/baeldung/streams/tomapvsgroupingby/ToMapVsGroupingByUnitTest.java (hunk 2 above)
Comment (theangrydev) [reply core-java-modules/core-java-streams-maps/src/test/java/com/baeldung/streams/tomapvsgroupingby/ToMapVsGroupingByUnitTest.java:72]: I agree, but technically we are supposed to be following https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names for constant naming, and the City type looks like an immutable class

I'm okay with lowercase final personally, not sure if the senior review will be picky about this or not

File: core-java-modules/core-java-streams-maps/src/test/java/com/baeldung/streams/tomapvsgroupingby/ToMapVsGroupingByUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Test;
+
+class City {
+
+    private String name;
+    private String country;
```
Comment (theangrydev) [inline core-java-modules/core-java-streams-maps/src/test/java/com/baeldung/streams/tomapvsgroupingby/ToMapVsGroupingByUnitTest.java:24]: And these can be final to emphasise that they do not change

File: core-java-modules/core-java-streams-maps/src/test/java/com/baeldung/streams/tomapvsgroupingby/ToMapVsGroupingByUnitTest.java (hunk 2 above)
Comment (sk1418) [reply core-java-modules/core-java-streams-maps/src/test/java/com/baeldung/streams/tomapvsgroupingby/ToMapVsGroupingByUnitTest.java:72]: Yes, technically they should be static variables. Ok, I will change them into "final static". 👍

**Review by theangrydev — APPROVED** [review]

**Pull request merged**